### DISCUSSION TITLE
Invalid copilot token: missing token: 403

### DISCUSSION THREAD

**Discussion by 0130w** [community/community#114909] [Copilot Conversations] [answered]
### Select Topic Area

Product Feedback

### Body

I use copilot from student package and I can't use it in these days. Error messages are as follows:
<img width="1198" alt="image" src="https://github.com/community/community/assets/106635619/6c86ddbf-9d0a-4bea-bfaf-c2f28c302ebb">

IDE: VS Code
OS: macOS Sonoma 14.4.1
Github username: 0130w

**Comment by 0130w** [comment]
<img width="1198" alt="image" src="https://github.com/community/community/assets/106635619/ee893246-a159-4261-bcc6-a980af18237e">
https://www.githubstatus.com/ says my copilot is all right.

**Reply by 0130w** [reply]
btw, same problem in IDEA copilot extension.

**Reply by AperturePlus** [reply]
same :(

**Comment by 0130w** [comment]
More info:

## Copilot

- Version: 1.176.0
- Build: prod
- Editor: vscode/1.87.2

## Environment

- http_proxy: n/a
- https_proxy: n/a
- no_proxy: n/a
- SSL_CERT_FILE: n/a
- SSL_CERT_DIR: n/a
- OPENSSL_CONF: n/a

## Feature Flags



## Node setup

- Number of root certificates: 137
- Operating system: Darwin
- Operating system version: 23.4.0
- Operating system architecture: arm64
- NODE_OPTIONS: n/a
- NODE_EXTRA_CA_CERTS: n/a
- NODE_TLS_REJECT_UNAUTHORIZED: n/a
- tls default min version: TLSv1.2
- tls default max version: TLSv1.3

## Network Configuration

- Proxy host: n/a
- Proxy port: n/a
- Kerberos SPN: n/a
- Reject unauthorized: disabled
- Fetcher: HelixFetcher

## Reachability

- github.com: HTTP 200 - 
- copilot-proxy.githubusercontent.com: HTTP 200 - 
- api.githubcopilot.com: HTTP 200 - 
- default.exp-tas.com: HTTP 200 - 

## VS Code Configuration

- HTTP proxy: 
- HTTP proxy authentication: n/a
- Proxy Strict SSL: true
- Extension HTTP proxy support: override

## Extensions

- Is `win-ca` installed?: false
- Is `mac-ca` installed?: false

## Authentication

- GitHub username: 0130w

**Comment by ha-h-ha** [comment]
Your access to Copilot has been restricted by our abuse detection systems. This is typically due to automated/excessive use, the use of custom clients to modify the Copilot service, or sharing credentials between multiple people.
 
We will not be able to reinstate your Copilot access, nor will you be able to have access to Copilot by purchasing it.
 
Please understand that repeated violations could result in your GitHub account being permanently suspended going forwards.

**Reply by 0130w** [reply]
Is this the response you received from community support? I don't think I've done anything illegal :|

**Reply by ha-h-ha** [reply]
I don't know if your reasons are the same as mine.But after I ask for  support, this is how they responded to me.

**Reply by AperturePlus** [reply]
so it's permanent?

**Reply by 0130w** [reply]
i don't know, I applied for a ticket and am waiting for a response.

**Reply by 0130w** [reply]
解封不了一点，我直接不用了

**Comment by 0130w** [comment, marked as answer]
The community has now restored my access to Copilot and informed me that the reason for this is that I'm using a Tencent Cloud agent to run Copilot. Now I'm checking where the problem might be.

**Reply by 0130w** [reply]
I think I should close this discussion now, I'll reopen it and reply when I find the possible cause.

**Reply by AperturePlus** [reply]
might be p0lit1cal cause. Plenty of cn mainland users encountered the same problem

**Reply by 0130w** [reply]
: - ( really annoying

**Comment by 0130w** [comment]
I'm currently unsure about the detection rules for blocking Copilot. My Copilot is currently back in use, but I'm not sure if I'll be able to use it consistently for some time to come. The community explained to me that I was banned because Copilot went through the Tencent Cloud proxy, but I don't know where I set up the Tencent Cloud proxy yet.

**Reply by Unicorn-V** [reply]
I have the same problem as you. The community also explained that I used TencentCloud, but I have no idea what it is?

**Reply by dsj320** [reply]
i got the same reply ,how  can I restore my accessibilty for copilot?

**Reply by 0130w** [reply]
May be apply for a ticket. https://support.github.com/

**Reply by dsj320** [reply]
i have applied for a ticket but the official did not restore my access for copilot,i don't think i have done anything against the rules.

**Reply by AperturePlus** [reply]
I've got a response yesterday but the official hasn't reinstate my access yet. I wonder what did u do after receiving a response.

**Reply by 0130w** [reply]
Officials reinstated Copilot for me after asking me to agree to the Copilot usage specification.

**Reply by dsj320** [reply]
What have you done?Officials did't reinstate Copilot for me though I have opend a ticket and explained the reason.

**Reply by 0130w** [reply]
Once again, my Copilot is banned. : )

**Reply by AperturePlus** [reply]
it reveals one thing: WWIII is coming.

**Comment by MC-dusk** [comment]
You must't use copilot directly in mainland even if your network can visit github directly. A proxy is necessary because github considers users' visit in mainland as an illegal behavior, which is so-called "excessive or automated usage and/or proxy hosting(TencentCloud)". A totally _badger game_.

**Reply by 0130w** [reply]
Thanks a lot, I'll be careful.

**Reply by Unicorn-V** [reply]
In other words, it is necessary to use proxy for Copilot? I understand, thank you.
